client/store/modules/comments/actions.js

```diff
@@ -1,14 +1,20 @@
+import find from 'lodash/find';
 import generateActions from '../../helpers/actions';
+import omitBy from 'lodash/omitBy';
 import SSEClient from '../../../SSEClient';
 
 const { api, get, save, setEndpoint, update } = generateActions();
 let SSE_CLIENT;
 
+const omitExisting = (fetched, existing) => {
+  return omitBy(fetched, it => find(existing, { id: it.id }));
+};
+
 const fetch = ({ state, commit }, { id, repositoryId }) => {
   const action = state.repositoryId === repositoryId ? 'fetch' : 'reset';
   if (action === 'reset') commit('setRepository', repositoryId);
   return api.fetch({ activityId: id }).then(items => {
-    commit(action, items);
+    commit(action, omitExisting(items, state.items));
```

Comment:
You can consider going step further and use differenceWith so you can inline it 😉 

What concerns me is that resource wrapper should set proper _cid for retrieved record if the record has been already mapped/retrieved (see resource within store helpers).  If this was properly set then it wouldn't be duplicated since items are stored within object keyed by _cid. So the answer should be that upon sseAdd (need to manage it within actions since you have api exposed there) you should use setCid function to create mapping 😉

Reply:
Thanks for the tip. There was indeed a deeper issue in the resource file. In the processEntries function we would fill the acc with all the items, even those whose cuid is already present (it would just get overwritten which in itself is not a problem the problem arises when you fill the acc with everything). Now we have a check there that prevents inserting into the accumulator. Please check @underscope